### ISSUE TITLE
Platform.sh CLI throws `segmentation fault` on M1/ARM with `db:dump` commmand

### ISSUE THREAD

**Issue by mattbloomfield** [docksal/docksal#1629]
**Description**

Problem/feature description...

I've just upgraded to an M1 Macbook and am trying to get set up. [This issue](https://github.com/docksal/docksal/issues/1421) has been extremely helpful and the environment is up. However, I have a custom command to sync my local database with the production database that is failing. 

This is the whole command: 

```shell
  echo "${YELLOW}DUMPING LOCAL DB TO /tmp/${MYSQL_DATABASE}_local_${TRANSACTION_ID}.sql${RESET}"
  # Dump existing local database (for safety)
  fin db dump /tmp/${MYSQL_DATABASE}_local_${TRANSACTION_ID}.sql

  echo "${YELLOW}DROPPING LOCAL DB${RESET}"
  # Delete local database if exists
  fin db cli "DROP DATABASE ${MYSQL_DATABASE}"

  echo "${YELLOW}CREATING LOCAL DB${RESET}"
  # Create new local database
  fin db create $MYSQL_DATABASE

  echo "${YELLOW}DUMPING REMOTE DB FROM ${REMOTE_ENV}${RESET}"
  # Dump the remote database and save locally
  fin platform db:dump --project $HOSTING_SITE --gzip -e $REMOTE_ENV -f /tmp/${MYSQL_DATABASE}_remote_${REMOTE_ENV}_${TRANSACTION_ID}.sql.gz

  echo "${YELLOW}IMPORTING REMOTE DB${RESET}"
  # Import locally saved database into container
  fin zcat < /tmp/${MYSQL_DATABASE}_remote_${REMOTE_ENV}_${TRANSACTION_ID}.sql.gz | fin exec "mysql -u root -p$MYSQL_ROOT_PASSWORD -h db $MYSQL_DATABASE"
```

Everything works fine until the `fin platform db:dump` command. At that point I get a segmentation fault. 

**Steps to reproduce the issue:**

1. Create a site based in platform.sh
2. Use the platform CLI built into the docksal containers to do a `fin platform db:dump -f /tmp/mydatabase.sql`

**Describe the results you received:**

`Segmentation Fault` 

**Describe the results you expected:**

It would save in the container's /tmp dir so I could import it. 

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
$ fin config                                                                                                                                             (git)-[arm-processor|merge]-
---------------------
COMPOSE_PROJECT_NAME_SAFE: uschamber
COMPOSE_FILE:
/Users/mbloomfield/.docksal/stacks/volumes-nfs.yml
/Users/mbloomfield/.docksal/stacks/stack-default.yml
/Users/mbloomfield/code/uscc/uschamber/.docksal/docksal.yml
ENV_FILE:
/Users/mbloomfield/code/uscc/uschamber/.docksal/docksal.env
/Users/mbloomfield/code/uscc/uschamber/.docksal/docksal-local.env

PROJECT_ROOT: /Users/mbloomfield/code/uscc/uschamber
DOCROOT: web
VIRTUAL_HOST: uschamber.docksal
VIRTUAL_HOST_ALIASES: *.uschamber.docksal
IP: 192.168.64.100
```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
$ fin sysinfo                                                                                                                                                (arm-processor)uschamber

███  FIN
fin version: 1.107.1

███  OS
Darwin macOS 12.2
Darwin wdcml9562.wireless.weber.edu 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000 arm64

███  ENVIRONMENT
MODE : Docker Desktop
DOCKER_HOST : 

███  NFS
DOCKSAL_NFS_PATH : /Users

nfsd service is enabled
nfsd is running (pid 245, 8 threads)

NFS EXPORTS
----------

# <ds-nfs docksal
/Users 127.0.0.1 192.168.64.1 -alldirs -maproot=0:0
# ds-nfs>
----------

Exports list on localhost:
/Users                              127.0.0.1 192.168.64.1

███  DOCKER
EXPECTED CLIENT VERSION: 20.10.7
EXPECTED SERVER VERSION: 20.10.7

Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.4
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:22 2021
 OS/Arch:           darwin/amd64 (rosetta)
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Mon Dec 13 11:43:07 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

███  DOCKER COMPOSE
EXPECTED VERSION: 1.29.2
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020

███  DOCKSAL: PROJECTS
project     STATUS                       virtual host                                                project root
co          Up About an hour (healthy)   co.docksal,*.co.docksal,co.docksal.*                        /Users/mbloomfield/code/uscc/co
uschamber   Up 4 hours (healthy)         uschamber.docksal,*.uschamber.docksal,uschamber.docksal.*   /Users/mbloomfield/code/uscc/uschamber

███  DOCKSAL: VIRTUAL HOSTS
*.co.docksal
*.uschamber.docksal
co.docksal.*
co.docksal
uschamber.docksal.*
uschamber.docksal

███  DOCKSAL: NETWORKING

DOCKSAL_IP: 192.168.64.100
DOCKSAL_HOST_IP: 192.168.64.1
DOCKSAL_VHOST_PROXY_IP: 0.0.0.0
DOCKSAL_DNS_IP: 0.0.0.0
DOCKSAL_DNS_DISABLED: 0
DOCKSAL_NO_DNS_RESOLVER: 0
DOCKSAL_DNS_UPSTREAM: 
DOCKSAL_DNS_DOMAIN: docksal

███  DOCKSAL: CONNECTIVITY

Host to 192.168.64.100:         PASS
Container to 192.168.64.100:    PASS
Container to 192.168.64.1:      PASS

Checking connectivity to http://dns-test.docksal...
Host: PASS
Containers: PASS

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID   IMAGE                     COMMAND                  CREATED             STATUS                       PORTS                                      NAMES
2c177080e5bd   docksal/apache:2.4-2.3    "httpd-foreground"       About an hour ago   Up About an hour (healthy)   80/tcp, 443/tcp                            co_web_1
8958d4db37c4   docksal/cli:php8.0-3.0    "/opt/startup.sh sup…"   About an hour ago   Up About an hour (healthy)   22/tcp, 3000/tcp, 9000/tcp                 co_cli_1
b900c44da588   docksal/vhost-proxy:1.7   "docker-entrypoint.s…"   3 hours ago         Up 3 hours (healthy)         0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   docksal-vhost-proxy
b4ddcc11042f   docksal/mariadb:10.3      "docker-entrypoint.s…"   3 hours ago         Up About an hour (healthy)   0.0.0.0:33062->3306/tcp                    co_db_1
adc56212291f   arm64v8/redis             "docker-entrypoint.s…"   3 hours ago         Up About an hour             6379/tcp                                   co_redis_1
f13aea6a024c   docksal/apache:2.4-2.3    "httpd-foreground"       4 hours ago         Up 4 hours (healthy)         80/tcp, 443/tcp                            uschamber_web_1
9ccdc8e8f079   docksal/cli:php8.0-3.0    "/opt/startup.sh sup…"   4 hours ago         Up 4 hours (healthy)         22/tcp, 3000/tcp, 9000/tcp                 uschamber_cli_1
a294499f7031   docksal/mariadb:10.3      "docker-entrypoint.s…"   4 hours ago         Up 4 hours (healthy)         0.0.0.0:33061->3306/tcp                    uschamber_db_1
4ac7e9dd7825   arm64v8/redis             "docker-entrypoint.s…"   4 hours ago         Up 4 hours                   6379/tcp                                   uschamber_redis_1
0862f210475e   docksal/ssh-agent:1.3     "docker-entrypoint.s…"   5 days ago          Up 5 days (healthy)                                                     docksal-ssh-agent
b309409a2fb8   docksal/dns:1.1           "docker-entrypoint.s…"   5 days ago          Up 5 days (unhealthy)        0.0.0.0:53->53/udp                         docksal-dns

███  DOCKER: NETWORKS
NETWORK ID     NAME                DRIVER    SCOPE
0ec49779294c   _default            bridge    local
c4954274118d   bridge              bridge    local
1d5e19350e65   co_default          bridge    local
5f328a365158   host                host      local
2fcb9e5eb3ab   none                null      local
ca98a1a253c9   uschamber_default   bridge    local

███  DOCKER DESKTOP
EXPECTED VERSION: 3.4.0
DETECTED VERSION: 4.4.2

███  HDD Usage
Filesystem       Size   Used  Avail Capacity iused       ifree %iused  Mounted on
/dev/disk3s1s1  1.8Ti   15Gi  1.7Ti     1%  577694  4293878363    0%   /
devfs           208Ki  208Ki    0Bi   100%     720           0  100%   /dev
/dev/disk3s6    1.8Ti   20Ki  1.7Ti     1%       0 18544762960    0%   /System/Volumes/VM
/dev/disk3s2    1.8Ti  277Mi  1.7Ti     1%     190 18544762960    0%   /System/Volumes/Preboot
/dev/disk3s4    1.8Ti  1.1Mi  1.7Ti     1%      41 18544762960    0%   /System/Volumes/Update
/dev/disk1s2    500Mi  6.0Mi  480Mi     2%       1     4911720    0%   /System/Volumes/xarts
/dev/disk1s1    500Mi  7.2Mi  480Mi     2%      14     4911720    0%   /System/Volumes/iSCPreboot
/dev/disk1s3    500Mi  2.3Mi  480Mi     1%      48     4911720    0%   /System/Volumes/Hardware
/dev/disk3s5    1.8Ti   74Gi  1.7Ti     4%  692902 18544762960    0%   /System/Volumes/Data
map auto_home     0Bi    0Bi    0Bi   100%       0           0  100%   /System/Volumes/Data/home
  ```

</details>

**Comment by lmakarov** [comment]
This should be addressed by https://github.com/docksal/service-cli/issues/265 and is available in the most recent docksal/cli [v3.1.0](https://github.com/docksal/service-cli/releases/tag/v3.1.0) images:

```
docksal/cli:php7.4-3.1
docksal/cli:php8.0-3.1
docksal/cli:php8.1-3.1
```

The upcoming Docksal v1.17 release will use these images.

**Comment by mattbloomfield** [comment]
Oh perfect, thanks! I was on `docksal/cli:php8.0-3.0`
